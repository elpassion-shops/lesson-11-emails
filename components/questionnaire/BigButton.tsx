
export function BigButton() {
    return (
      <button
        type="submit"
        className={"confirm-button bg-rose-600 px-4 py-2 rounded-md text-white drop-shadow-md"}
      >
        Wy≈õlij odpowiedzi
      </button>
    );
  }
  