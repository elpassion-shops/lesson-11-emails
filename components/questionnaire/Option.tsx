import { useState } from "react";

interface OptionProps {
  answear: number;
  className: string;
  onClick(): void;
}

export function Option(props: OptionProps) {
  return (
    <button
      className={props.className}
      onClick={() => {
        props.onClick();
      }}
    >
      {props.answear}
    </button>
  );
}
