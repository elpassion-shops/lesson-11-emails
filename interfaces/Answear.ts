export interface Answear {
  rating: number;
  key: number;
}
