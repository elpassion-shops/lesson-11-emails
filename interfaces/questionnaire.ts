import {
  IsEmail,
  IsInt,
  IsNotIn,
  IsOptional,
  IsString,
  Max,
  Min,
  ValidateNested,
} from "class-validator";

export interface IAnswer {
  isOpen: boolean;
  value?: number | string | null;
  questionId?: number;

}

export interface IQuestion {
  id: number;
  question: string;
  answer?: IAnswer;
  isOpen?: boolean | null;
}

export interface IQuestionnaire {
  id: number;
  title: string;
  questions: IQuestion[];
  email?: string | null;
}

export interface IQuestionnaireResponse {
  questionnaire: IQuestionnaire;
}

export interface IQuestionnaireRequest {
  id: number;
}

export interface IAnswersRequest {
  answers: IAnswer[];
  email: string;
}

export class Answer implements IAnswer{
  isOpen: boolean;
  questionId: number;
  @IsNotIn([0,"",null,undefined])
  value: string | number;

}

export class AnswerRequest {
  @ValidateNested({each: true})
  answers: Answer[];

  @IsEmail()
  email: string;
}

export class AnswerClose implements IAnswer {
  isOpen = false;

  @IsInt()
  @Min(1)
  @Max(5)
  value?: number;

  constructor(value: number) {
    this.value = value;
  }
}

export class AnswerOpen implements IAnswer {
  isOpen = true;

  @IsString()
  value?: string;

  constructor(value: string) {
    this.value = value;
  }
}

export class Question implements IQuestion {
  @IsInt()
  @Min(1)
  id: number;

  @IsString()
  question: string;

  @ValidateNested()
  answer: IAnswer;

  constructor(id: number, question: string, answer: IAnswer) {
    this.id = id;
    this.question = question;
    this.answer = answer;
  }
}

export class Questionnaire implements IQuestionnaire {
  @IsInt()
  @Min(1)
  id;

  @IsString()
  title;

  @ValidateNested({ each: true })
  questions;

  @IsOptional()
  @IsEmail()
  email?;

  constructor(
    id: number,
    title: string,
    questions: IQuestion[],
    email?: string | null
  ) {
    this.id = id;
    this.title = title;
    this.questions = questions;
    this.email = email;
  }
}
