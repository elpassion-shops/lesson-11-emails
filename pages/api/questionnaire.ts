import "reflect-metadata";
import { validate, validateOrReject } from "class-validator";
import type { NextApiRequest, NextApiResponse } from "next";
import { Request } from "../../dtos/request";
import { plainToClass } from "class-transformer";

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse
) {
  try {
    const data = req.body;
    const request = plainToClass(Request, data);
    await validateOrReject(request);
    console.log(request);
    res.status(201).json({});
  } catch (error) {
    res.status(400).json(error);
  }
}
