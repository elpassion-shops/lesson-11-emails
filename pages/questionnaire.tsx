import { useRouter } from "next/router";
import React from "react";
import ReactDOM from "react-dom";
import { useForm, SubmitHandler } from "react-hook-form";

enum GenderEnum {
    female = "female",
    male = "male",
    other = "other"
  }
  interface IFormInput {
    firstName: String;
    gender: GenderEnum;
  }
  

//const resolver = classValidatorResolver
function Questionnaire() {

  const { register, handleSubmit } = useForm<IFormInput>();
  const onSubmit = (data: IFormInput) => console.log(data);

  return (
    <form >
      <label>First Name</label>
      <input {...register("firstName")} />
      <label>Gender Selection</label>
      <select {...register("gender")}>
        <option value="female">female</option>
        <option value="male">male</option>
        <option value="other">other</option>
      </select>
      <input type="submit" />
    </form>
  );
}

  export default Questionnaire;
  