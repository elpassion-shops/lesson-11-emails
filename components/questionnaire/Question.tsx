import { useEffect, useState } from "react";
import { Answear } from "../../interfaces/Answear";
import { Option } from "./Option";

interface QuestionProps {
  text: string;
  hint: string;
  answears: Answear[];
  selected: number;
  onClick(answear: number): void;
}

export function Question(props: QuestionProps) {
  return (
    <div className="question">
      <p>{props.text}</p>
      <p className="hint">{props.hint}</p>
      <div className="options">
        {props.answears.map((a) => {
          return (
            <Option
              className={props.selected === a.rating? "selected" : "option-button"}
              onClick={() => {
                props.onClick(a.rating);
              }}
              answear={a.rating}
              key={a.key}
            />
          );
        })}
      </div>
    </div>
  );
}
