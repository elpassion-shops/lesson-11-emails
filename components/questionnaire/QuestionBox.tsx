import { useState } from "react";
import { Answear } from "../../interfaces/Answear";
import { Question } from "./Question";

export function QuestionBox() {
  interface QuestionWithAnswear {
    question: string;
    answear: number;
  }

  interface Question {
    question: string;
    hint: string;
    key: number;
    selected: number;
    answears: Answear[];
  }

  const [questions, setQuestions] = useState<Question[]>([
    {
      question:
        "Jak podobał Ci się nasz staż? (ostatnia szansa na zmianę zdania! :D)",
      hint: "(1 - w ogóle, 2 - niezbyt, 3 - średnio, 4 - był ok, 5 - bardzo)",
      key: Math.random(),
      selected: 0,
      answears: [
        { rating: 1, key: Math.random()},
        { rating: 2, key: Math.random()},
        { rating: 3, key: Math.random()},
        { rating: 4, key: Math.random()},
        { rating: 5, key: Math.random()}
      ],
    },
    {
      question: "Jak dobrze był Twoim zdaniem przygotowany prowadzący?",
      hint: "(1 - fatalnie, 2 - kiepsko, 3 - średnio, 4 - był ok, 5 - świetnie",
      key: Math.random(),
      selected: 0,
      answears: [
        { rating: 1, key: Math.random()},
        { rating: 2, key: Math.random()},
        { rating: 3, key: Math.random()},
        { rating: 4, key: Math.random()},
        { rating: 5, key: Math.random()},
      ],
    },
    {
      question: "Czy uważasz, że zdobyta wiedza przyda Ci się w życiu?",
      hint: "(1 - w ogóle mi się nie przyda, 2 - raczej nie, 3 - może, 4 - myślę, że tak, 5 - na pewno!",
      key: Math.random(),
      selected: 0,
      answears: [
        { rating: 1, key: Math.random()},
        { rating: 2, key: Math.random()},
        { rating: 3, key: Math.random()},
        { rating: 4, key: Math.random()},
        { rating: 5, key: Math.random()}
      ],
    },
  ]);

  const [questionsWithAnswears, setQuestionsWithAnswers] = useState<
    QuestionWithAnswear[]
  >([]);

  function handleClick(question: string, answear: number) {
    setQuestionsWithAnswers((prevState) =>
      prevState.filter((q) => q.question !== question)
    );
    setQuestionsWithAnswers((prevState) =>
      prevState.concat({ question: question, answear: answear })
    );

    setQuestions((prevState) => {
      return prevState.map((q) => {
        if (q.question !== question) return q;
        return {question: q.question, hint: q.hint, key: q.key, selected: answear, answears: q.answears}
        })
  })
}
  return (
    <div className="question-box">
      {questions.map((q) => {
        return (
          <Question
            onClick={(answear) => handleClick(q.question, answear)}
            text={q.question}
            hint={q.hint}
            key={q.key}
            answears={q.answears}
            selected={q.selected}
          />
        );
      })}
    </div>
  );
}
