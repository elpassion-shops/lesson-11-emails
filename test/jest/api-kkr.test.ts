import {
  IsBoolean,
  IsInt,
  IsString,
  Max,
  Min,
  validate,
  ValidateNested,
} from 'class-validator';
import { IAnswer, IQuestion } from '../../interfaces/questionnaire';

class AnswerOpen implements IAnswer {
  @IsBoolean()
  isOpen!: boolean;

  @IsString()
  value!: string;
}

function generateAnswerOpen(overrides: { isOpen?: any; value?: any }) {
  const answerOpen = new AnswerOpen();
  answerOpen.value =
    overrides.value === undefined ? 'very good' : overrides.value;
  answerOpen.isOpen = overrides.isOpen === undefined ? true : overrides.isOpen;
  return answerOpen;
}

class AnswerClose implements IAnswer {
  @IsBoolean()
  isOpen!: boolean;

  @Min(1)
  @Max(5)
  @IsInt()
  value!: number;
}

function generateAnswerClose(overrides: { isOpen?: any; value?: any }) {
  const answerClose = new AnswerClose();
  answerClose.value = overrides.value === undefined ? 5 : overrides.value;
  answerClose.isOpen = overrides.isOpen === undefined ? true : overrides.isOpen;
  return answerClose;
}

let answerCloseinvalid = generateAnswerClose({ value: null });

describe('tests for AnswerOpen', () => {
  const answerOpenValid = generateAnswerOpen({});

  describe('tests for AnswerOpen.value', () => {
    test('should pass with string in value', async () => {
      expect(await validate(answerOpenValid)).toEqual([]);
    });

    test('should fails with empty value', async () => {
      const answerOpenInvalid = generateAnswerOpen({ value: null });
      expect(await validate(answerOpenInvalid)).toMatchSnapshot();
    });

    test('should fails with number in value', async () => {
      const answerOpenInvalid = generateAnswerOpen({ value: 9 });
      expect(await validate(answerOpenInvalid)).toMatchSnapshot();
    });

    test('should fails with boolean in value', async () => {
      const answerOpenInvalid = generateAnswerOpen({ value: true });
      expect(await validate(answerOpenInvalid)).toMatchSnapshot();
    });
  });

  describe('tests for AnswerOpen.isOpen', () => {
    test('should pass with boolean in isOpen', async () => {
      expect(await validate(answerOpenValid)).toEqual([]);
    });

    test('should fails with empty isOpen', async () => {
      const answerOpenInvalid = generateAnswerOpen({ isOpen: null });
      expect(await validate(answerOpenInvalid)).toMatchSnapshot();
    });

    test('should fails with string in isOpen', async () => {
      const answerOpenInvalid = generateAnswerOpen({ isOpen: 'abc' });
      expect(await validate(answerOpenInvalid)).toMatchSnapshot();
    });

    test('should fails with number in isOpen', async () => {
      const answerOpenInvalid = generateAnswerOpen({ isOpen: 9 });
      expect(await validate(answerOpenInvalid)).toMatchSnapshot();
    });
  });
});

describe('tests for AnswerClose', () => {
  const answerCloseValid = generateAnswerClose({});

  describe('tests for AnswerClose.value', () => {
    test('should pass with correct number in value', async () => {
      expect(await validate(answerCloseValid)).toEqual([]);
    });
    test('should fails with to low value', async () => {
      const answerCloseInvalid = generateAnswerClose({ value: 0 });
      expect(await validate(answerCloseInvalid)).toMatchSnapshot();
    });

    test('should fails with to hight value', async () => {
      const answerCloseInvalid = generateAnswerClose({ value: 6 });
      expect(await validate(answerCloseInvalid)).toMatchSnapshot();
    });

    test('should fails with empty value', async () => {
      const answerCloseInvalid = generateAnswerClose({ value: null });
      expect(await validate(answerCloseInvalid)).toMatchSnapshot();
    });

    test('should fails with number in value', async () => {
      const answerCloseInvalid = generateAnswerClose({ value: 9 });
      expect(await validate(answerCloseInvalid)).toMatchSnapshot();
    });

    test('should fails with boolean in value', async () => {
      const answerCloseInvalid = generateAnswerClose({ value: true });
      expect(await validate(answerCloseInvalid)).toMatchSnapshot();
    });
  });

  describe('tests for AnswerClose.isOpen', () => {
    test('should pass with boolean in isOpen', async () => {
      expect(await validate(answerCloseValid)).toEqual([]);
    });

    test('should fails with empty isOpen', async () => {
      const answerCloseInvalid = generateAnswerClose({ isOpen: null });
      expect(await validate(answerCloseInvalid)).toMatchSnapshot();
    });

    test('should fails with string in isOpen', async () => {
      const answerCloseInvalid = generateAnswerClose({ isOpen: 'abc' });
      expect(await validate(answerCloseInvalid)).toMatchSnapshot();
    });

    test('should fails with number in isOpen', async () => {
      const answerCloseInvalid = generateAnswerClose({ isOpen: 9 });
      expect(await validate(answerCloseInvalid)).toMatchSnapshot();
    });
  });
});

describe('tests for Question', () => {
  class Question implements IQuestion {
    @IsInt()
    @Min(1)
    id!: number;

    @IsString()
    question!: string;

    @ValidateNested()
    answer?: IAnswer | null;
  }

  function generateQuestion(overrides: {
    id?: any;
    question?: any;
    answer?: IAnswer | any;
  }) {
    const answerOpenValid = generateAnswerOpen({});

    const question = new Question();
    question.id = overrides.id === undefined ? 1 : overrides.id;
    question.question =
      overrides.question === undefined ? 'How about ?' : overrides.question;
    question.answer =
      overrides.answer === undefined ? answerOpenValid : overrides.answer;
    return question;
  }

  const questionValid = generateQuestion({});

  describe('tests for Question.id', () => {
    test('should pass with correct number in id', async () => {
      expect(await validate(questionValid)).toEqual([]);
    });

    test('should fails with to low value', async () => {
      const questionInvalid = generateQuestion({ id: 0 });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });

    test('should fails with empty value', async () => {
      const questionInvalid = generateQuestion({ id: null });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });

    test('should fails with boolean in id', async () => {
      const questionInvalid = generateQuestion({ id: true });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });
    test('should fails with text in id', async () => {
      const questionInvalid = generateQuestion({ id: 'text w id' });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });
  });

  describe('tests for Question.question', () => {
    test('should pass with string in question', async () => {
      expect(await validate(questionValid)).toEqual([]);
    });

    test('should fails with empty question', async () => {
      const questionInvalid = generateQuestion({ question: null });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });

    test('should fails with number in question', async () => {
      const questionInvalid = generateQuestion({ question: 9 });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });

    test('should fails with boolean in question', async () => {
      const questionInvalid = generateQuestion({ question: true });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });
  });

  describe('tests for Question.answer', () => {
    test('should pass with correct answer', async () => {
      expect(await validate(questionValid)).toEqual([]);
    });

    test('should fails with empty answer', async () => {
      const questionInvalid = generateQuestion({ answer: null });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });

    test('should fails with string in answer', async () => {
      const questionInvalid = generateQuestion({ answer: 'uno duo quarto' });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });

    test('should fails with number in answer', async () => {
      const questionInvalid = generateQuestion({ answer: 9 });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });

    test('should fails with boolean in answer', async () => {
      const questionInvalid = generateQuestion({ answer: true });
      expect(await validate(questionInvalid)).toMatchSnapshot();
    });
  });
});
