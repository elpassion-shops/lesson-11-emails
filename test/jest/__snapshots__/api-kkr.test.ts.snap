// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for AnswerClose tests for AnswerClose.isOpen should fails with empty isOpen 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isBoolean": "isOpen must be a boolean value",
    },
    "property": "isOpen",
    "target": AnswerClose {
      "isOpen": null,
      "value": 5,
    },
    "value": null,
  },
]
`;

exports[`tests for AnswerClose tests for AnswerClose.isOpen should fails with number in isOpen 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isBoolean": "isOpen must be a boolean value",
    },
    "property": "isOpen",
    "target": AnswerClose {
      "isOpen": 9,
      "value": 5,
    },
    "value": 9,
  },
]
`;

exports[`tests for AnswerClose tests for AnswerClose.isOpen should fails with string in isOpen 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isBoolean": "isOpen must be a boolean value",
    },
    "property": "isOpen",
    "target": AnswerClose {
      "isOpen": "abc",
      "value": 5,
    },
    "value": "abc",
  },
]
`;

exports[`tests for AnswerClose tests for AnswerClose.value should fails with boolean in value 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isInt": "value must be an integer number",
      "max": "value must not be greater than 5",
      "min": "value must not be less than 1",
    },
    "property": "value",
    "target": AnswerClose {
      "isOpen": true,
      "value": true,
    },
    "value": true,
  },
]
`;

exports[`tests for AnswerClose tests for AnswerClose.value should fails with empty value 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isInt": "value must be an integer number",
      "max": "value must not be greater than 5",
      "min": "value must not be less than 1",
    },
    "property": "value",
    "target": AnswerClose {
      "isOpen": true,
      "value": null,
    },
    "value": null,
  },
]
`;

exports[`tests for AnswerClose tests for AnswerClose.value should fails with number in value 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "max": "value must not be greater than 5",
    },
    "property": "value",
    "target": AnswerClose {
      "isOpen": true,
      "value": 9,
    },
    "value": 9,
  },
]
`;

exports[`tests for AnswerClose tests for AnswerClose.value should fails with to hight value 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "max": "value must not be greater than 5",
    },
    "property": "value",
    "target": AnswerClose {
      "isOpen": true,
      "value": 6,
    },
    "value": 6,
  },
]
`;

exports[`tests for AnswerClose tests for AnswerClose.value should fails with to low value 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "min": "value must not be less than 1",
    },
    "property": "value",
    "target": AnswerClose {
      "isOpen": true,
      "value": 0,
    },
    "value": 0,
  },
]
`;

exports[`tests for AnswerOpen tests for AnswerOpen.isOpen should fails with empty isOpen 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isBoolean": "isOpen must be a boolean value",
    },
    "property": "isOpen",
    "target": AnswerOpen {
      "isOpen": null,
      "value": "very good",
    },
    "value": null,
  },
]
`;

exports[`tests for AnswerOpen tests for AnswerOpen.isOpen should fails with number in isOpen 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isBoolean": "isOpen must be a boolean value",
    },
    "property": "isOpen",
    "target": AnswerOpen {
      "isOpen": 9,
      "value": "very good",
    },
    "value": 9,
  },
]
`;

exports[`tests for AnswerOpen tests for AnswerOpen.isOpen should fails with string in isOpen 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isBoolean": "isOpen must be a boolean value",
    },
    "property": "isOpen",
    "target": AnswerOpen {
      "isOpen": "abc",
      "value": "very good",
    },
    "value": "abc",
  },
]
`;

exports[`tests for AnswerOpen tests for AnswerOpen.value should fails with boolean in value 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "value must be a string",
    },
    "property": "value",
    "target": AnswerOpen {
      "isOpen": true,
      "value": true,
    },
    "value": true,
  },
]
`;

exports[`tests for AnswerOpen tests for AnswerOpen.value should fails with boolean in value 2`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "value must be a string",
    },
    "property": "value",
    "target": AnswerOpen {
      "isOpen": true,
      "value": true,
    },
    "value": true,
  },
]
`;

exports[`tests for AnswerOpen tests for AnswerOpen.value should fails with empty value 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "value must be a string",
    },
    "property": "value",
    "target": AnswerOpen {
      "isOpen": true,
      "value": null,
    },
    "value": null,
  },
]
`;

exports[`tests for AnswerOpen tests for AnswerOpen.value should fails with empty value 2`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "value must be a string",
    },
    "property": "value",
    "target": AnswerOpen {
      "isOpen": true,
      "value": null,
    },
    "value": null,
  },
]
`;

exports[`tests for AnswerOpen tests for AnswerOpen.value should fails with number in value 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "value must be a string",
    },
    "property": "value",
    "target": AnswerOpen {
      "isOpen": true,
      "value": 9,
    },
    "value": 9,
  },
]
`;

exports[`tests for AnswerOpen tests for AnswerOpen.value should fails with number in value 2`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "value must be a string",
    },
    "property": "value",
    "target": AnswerOpen {
      "isOpen": true,
      "value": 9,
    },
    "value": 9,
  },
]
`;

exports[`tests for Question tests for Question.answer should fails with boolean in answer 1`] = `
Array [
  ValidationError {
    "children": Array [
      ValidationError {
        "constraints": Object {
          "nestedValidation": "nested property answer must be either object or array",
        },
        "property": "answer",
        "target": [Function],
        "value": true,
      },
    ],
    "property": "answer",
    "target": Question {
      "answer": true,
      "id": 1,
      "question": "How about ?",
    },
    "value": true,
  },
]
`;

exports[`tests for Question tests for Question.answer should fails with boolean in answer 2`] = `Array []`;

exports[`tests for Question tests for Question.answer should fails with empty answer 1`] = `
Array [
  ValidationError {
    "children": Array [
      ValidationError {
        "constraints": Object {
          "nestedValidation": "nested property answer must be either object or array",
        },
        "property": "answer",
        "target": [Function],
        "value": null,
      },
    ],
    "property": "answer",
    "target": Question {
      "answer": null,
      "id": 1,
      "question": "How about ?",
    },
    "value": null,
  },
]
`;

exports[`tests for Question tests for Question.answer should fails with number in answer 1`] = `
Array [
  ValidationError {
    "children": Array [
      ValidationError {
        "constraints": Object {
          "nestedValidation": "nested property answer must be either object or array",
        },
        "property": "answer",
        "target": [Function],
        "value": 9,
      },
    ],
    "property": "answer",
    "target": Question {
      "answer": 9,
      "id": 1,
      "question": "How about ?",
    },
    "value": 9,
  },
]
`;

exports[`tests for Question tests for Question.answer should fails with string in answer 1`] = `
Array [
  ValidationError {
    "children": Array [
      ValidationError {
        "constraints": Object {
          "nestedValidation": "nested property answer must be either object or array",
        },
        "property": "answer",
        "target": [Function],
        "value": "uno duo quarto",
      },
    ],
    "property": "answer",
    "target": Question {
      "answer": "uno duo quarto",
      "id": 1,
      "question": "How about ?",
    },
    "value": "uno duo quarto",
  },
]
`;

exports[`tests for Question tests for Question.id should fails with boolean in id 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isInt": "id must be an integer number",
      "min": "id must not be less than 1",
    },
    "property": "id",
    "target": Question {
      "answer": AnswerOpen {
        "isOpen": true,
        "value": "very good",
      },
      "id": true,
      "question": "How about ?",
    },
    "value": true,
  },
]
`;

exports[`tests for Question tests for Question.id should fails with boolean in id 2`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isInt": "id must be an integer number",
      "min": "id must not be less than 1",
    },
    "property": "id",
    "target": Question {
      "answer": AnswerOpen {
        "isOpen": true,
        "value": "very good",
      },
      "id": "text w id",
      "question": "How about ?",
    },
    "value": "text w id",
  },
]
`;

exports[`tests for Question tests for Question.id should fails with empty value 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isInt": "id must be an integer number",
      "min": "id must not be less than 1",
    },
    "property": "id",
    "target": Question {
      "answer": AnswerOpen {
        "isOpen": true,
        "value": "very good",
      },
      "id": null,
      "question": "How about ?",
    },
    "value": null,
  },
]
`;

exports[`tests for Question tests for Question.id should fails with text in id 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isInt": "id must be an integer number",
      "min": "id must not be less than 1",
    },
    "property": "id",
    "target": Question {
      "answer": AnswerOpen {
        "isOpen": true,
        "value": "very good",
      },
      "id": "text w id",
      "question": "How about ?",
    },
    "value": "text w id",
  },
]
`;

exports[`tests for Question tests for Question.id should fails with to low value 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "min": "id must not be less than 1",
    },
    "property": "id",
    "target": Question {
      "answer": AnswerOpen {
        "isOpen": true,
        "value": "very good",
      },
      "id": 0,
      "question": "How about ?",
    },
    "value": 0,
  },
]
`;

exports[`tests for Question tests for Question.question should fails with boolean in question 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "question must be a string",
    },
    "property": "question",
    "target": Question {
      "answer": AnswerOpen {
        "isOpen": true,
        "value": "very good",
      },
      "id": 1,
      "question": true,
    },
    "value": true,
  },
]
`;

exports[`tests for Question tests for Question.question should fails with empty question 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "question must be a string",
    },
    "property": "question",
    "target": Question {
      "answer": AnswerOpen {
        "isOpen": true,
        "value": "very good",
      },
      "id": 1,
      "question": null,
    },
    "value": null,
  },
]
`;

exports[`tests for Question tests for Question.question should fails with number in question 1`] = `
Array [
  ValidationError {
    "children": Array [],
    "constraints": Object {
      "isString": "question must be a string",
    },
    "property": "question",
    "target": Question {
      "answer": AnswerOpen {
        "isOpen": true,
        "value": "very good",
      },
      "id": 1,
      "question": 9,
    },
    "value": 9,
  },
]
`;
